elichika/elichika/parser/functions.py

```diff
@@ -366,10 +367,10 @@ def __init__(self, func):
         self.lineno = sourcelines[1]
 
         code = utils.clip_head(inspect.getsource(func))
-
+        processed_ast = canonicalizer.Canonicalizer().visit(ast.parse(code))
```

Comment:
I'm not sure, but maybe better to keep `gast.ast_to_gast` before the canonicalization? Otherwise, the canonicalizer may need to take care of multiple python versions? It seems gast also has NodeTransformer so I'm guessing we can use it.